testing/test_tmpdir.py

```diff
@@ -435,10 +435,23 @@ def test(tmp_path):
             fn.write_text('hello')
             mode = os.stat(str(fn)).st_mode
             os.chmod(str(fn), mode & ~stat.S_IREAD)
-    """
+        """
     )
     result = testdir.runpytest("--basetemp=tmp")
     assert result.ret == 0
     # running a second time and ensure we don't crash
     result = testdir.runpytest("--basetemp=tmp")
     assert result.ret == 0
+
+
+def test_basetemp_with_not_executable_dirs(testdir):
+    """Integration test for #7933"""
+    testdir.makepyfile(
+        """
+        def test(tmp_path):
+            (tmp_path / "foo").mkdir(mode=0)
```

Comment:
something to test recursively non-executable dirs

```suggestion
            p = tmp_path / "foo" / "bar" / "baz"
            p.parent.mkdir(parents=True)
            p.touch(mode=0)
            for p in p.parents:
                if p == tmp_path:
                    break
                p.chmod(mode=0)
```